src/browser/vdiff.js

```diff
@@ -1,28 +1,106 @@
-import { chai, expect } from '@open-wc/testing';
+import { chai, expect, nextFrame } from '@open-wc/testing';
 import { executeServerCommand } from '@web/test-runner-commands';
 
-let test;
+let test, soonPromise;
 
-chai.Assertion.addMethod('golden', function(...args) {
-	return ScreenshotAndCompare.call({ test, elem: this._obj }, ...args); // eslint-disable-line no-invalid-this
+/* eslint-disable no-undef, no-invalid-this */
+chai.Assertion.addMethod('golden', async function(...args) {
+	await soonPromise?.catch(err => expect.fail(err));
+	return ScreenshotAndCompare.call({ test, elem: this._obj }, ...args);
 });
-mocha.setup({ // eslint-disable-line no-undef
+
+chai.Assertion.addChainableMethod('soon',
+	() => { throw new TypeError('"soon" is not a function'); },
+
+	async function() {
+		let resolve, reject;
+		soonPromise = new Promise((res, rej) => {
+			resolve = res;
+			reject = rej;
+		});
+
+		try {
+			expect(this._obj).to.be.instanceof(HTMLElement);
+		} catch (e) {
+			reject(e.message);
+		}
+
+		await nextFrame();
+		await this._obj.updateComplete;
+		resolve();
+	}
+);
+
+mocha.setup({
 	rootHooks: {
 		beforeEach() {
 			test = this.currentTest;
 		}
 	}
 });
+/* eslint-enable */
 
 async function ScreenshotAndCompare(opts) {
+	await document.fonts.ready; // firefox fonts
+
+	if (window.d2lTest) {
+		inlineStyles(this.elem);
+		document.documentElement.classList.add('screenshot');
+	}
+
 	const name = this.test.fullTitle();
 	const rect = this.elem.getBoundingClientRect();
 	let result = await executeServerCommand('brightspace-visual-diff-compare', { name, rect, opts });
 	if (result.resizeRequired) {
 		this.test.timeout(0);
 		result = await executeServerCommand('brightspace-visual-diff-compare-resize', { name });
 	}
+
+	if (window.d2lTest) document.documentElement.classList.remove('screenshot');
+
 	if (!result.pass) {
+		if (window.d2lTest?.pause) {
+			window.d2lTest.fail();
+			await window.d2lTest.retryResponse;
+		}
 		expect.fail(result.message);
 	}
+
+	window.d2lTest?.pass();
+}
+
+const disallowedProps = ['width', 'inline-size'];
+let count = 0;
+function inlineStyles(elem) {
+	// headed chrome takes screenshots by first moving the element, which
+	// breaks the hover state. So, copy current styles inline before screenshot
```

Comment:
Whoa, crazy that we'd need to do this. I'm sure you tried this, but instead of making all the styles inline, could we just try to re-apply the hover after Chrome moves it?

Reply:
Yeah, it's not _just_ that it moves. It seems to be locking it down in some way, which makes sense, but it just loses the hover.